src/platforms/rcore_ios.c

```diff
@@ -0,0 +1,822 @@
+{
+    sync_all_touches(event);
+    send_gesture_event(touches, TOUCH_ACTION_MOVE);
+}
+- (void)touchesCancelled:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
+{
+    sync_all_touches(event);
+    send_gesture_event(touches, TOUCH_ACTION_CANCEL);
+}
+
+@end
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    // Override point for customization after application launch.
+    self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+    self.window.backgroundColor = [UIColor redColor];
+    self.window.rootViewController = [[GameViewController alloc] init];
+    [self.window makeKeyAndVisible];
+    ios_ready();
+    CADisplayLink *displayLink = [CADisplayLink displayLinkWithTarget:self.window.rootViewController selector:@selector(update)];
+    [displayLink addToRunLoop:[NSRunLoop currentRunLoop] forMode:NSDefaultRunLoopMode];
+    return YES;
+}
+
+- (void)applicationWillResignActive:(UIApplication *)application {
+    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
+    // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
+    CORE.Window.flags &= ~FLAG_WINDOW_UNFOCUSED;
+}
+- (void)applicationDidBecomeActive:(UIApplication *)application {
+    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+    CORE.Window.flags |= FLAG_WINDOW_UNFOCUSED;
+}
+- (void)applicationDidEnterBackground:(UIApplication *)application {
+    // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
+}
+
+- (void)applicationWillEnterForeground:(UIApplication *)application {
+    // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
+}
+
+- (void)applicationWillTerminate:(UIApplication *)application {
+    ios_destroy();
+
+    if (platform.device != EGL_NO_DISPLAY)
+    {
+        // the user does not call CloseWindow() before exiting
+        TRACELOG(LOG_ERROR, "DISPLAY: CloseWindow() should be called before terminating the application");
+    }
+    // If 'platform.device' is already set to 'EGL_NO_DISPLAY'
+    // this means that the user has already called 'CloseWindow()'
+}
+
+@end
+
+/* main() */
+int main(int argc, char * argv[]) {
+    return UIApplicationMain(argc, argv, nil, NSStringFromClass([AppDelegate class]));
```

Comment:
I imagine this is the biggest issue with iOS implementation, isn't there another alternative?